packages/twenty-front/src/modules/object-record/record-field/hooks/useIsRecordFieldReadOnly.ts

```diff
@@ -0,0 +1,57 @@
+import {
+  IsFieldReadOnlyByPermissionsParams,
+  useFieldIsReadOnlyByPermissions,
+} from '@/object-record/record-field/hooks/useIsFieldReadOnlyByPermissions';
+import { useIsRecordDeleted } from '@/object-record/record-field/hooks/useIsRecordDeleted';
+import {
+  isFieldReadOnlyBySystem,
+  IsFieldReadOnlyBySystemParams,
+} from '@/object-record/record-field/utils/isFieldReadOnlyBySystem';
+
+export const isRecordFieldReadOnly = ({
+  isRecordDeleted,
+  isFieldReadOnlyByPermissions,
+  isFieldReadOnlyBySystem,
+}: {
+  isRecordDeleted?: boolean;
+  isFieldReadOnlyByPermissions?: boolean;
+  isFieldReadOnlyBySystem?: boolean;
+}) => {
+  return (
+    (isRecordDeleted ||
+      isFieldReadOnlyByPermissions ||
+      isFieldReadOnlyBySystem) ??
+    false
+  );
+};
+
+export const useIsRecordFieldReadOnly = ({
+  recordId,
```

Comment:
revert to an interface same as useIsFieldValueReadonly

Reply:
this hook should have the whole information in its props. and just call the isFieldReadonly util